### ISSUE TITLE
⛔ 🐞 Bug Report: The Preview Changes from the Top Bar does not work inside the Theme Builder

### ISSUE THREAD

**Issue by awakekat** [elementor/elementor#21831]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Both Elementor and Elementor Pro are installed at 3.12
When in the Theme Builder the Preview Button does not work. If in the Theme Builder, we are not supposed to preview our parts, then why is the button there? Just removing the link from the button is not a solution, removing the button completely would solve the problem. I have switched between, regular pages and using the preview button and having it work properly. So if it is your intention to not allow the Preview Button to function while in the Theme Builder - remove it from the UI.

### Steps to reproduce

1) Go to the Theme Builder and open a part. Click on the Preview button - nothing happens.
2) Open a page and edit it with Elementor. Click the Preview button - it opens a new tab an shows the page.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt


					
== Server Environment ==
	Operating System: Darwin
	Software: nginx/1.16.0
	MySQL version: MySQL Community Server - GPL v8.0.16
	PHP Version: 7.4.1
	PHP Memory Limit: 256M
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2
	Site URL: http://chat.local
	Home URL: http://chat.local
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: info@elewebdev.com
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.7.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.12.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.12.0
		Author: Elementor.com


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Active
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Inactive
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Inactive
	Editor Top Bar: Active
	Landing Pages: Active
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Page Transitions: Inactive
	Notes: Inactive
	Loop: Active
	Scroll Snap: Inactive
	Menu: Active


== Log ==
	
Log: showing 20 of 342023-03-22 14:15:40 [info] elementor::elementor_updater Started 
2023-03-22 14:15:40 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-22 14:15:40 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-22 14:15:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.2',
  'to' => '3.11.5',
)]
2023-03-22 14:15:40 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.2',
  'to' => '3.11.5',
)]
2023-03-22 14:16:58 [info] elementor-pro::elementor_pro_updater Started 
2023-03-22 14:16:58 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-22 14:16:58 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-22 14:16:58 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.1',
  'to' => '3.11.7',
)]
2023-03-22 14:16:58 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.1',
  'to' => '3.11.7',
)]
2023-03-30 00:39:07 [info] elementor::elementor_updater Started 
2023-03-30 00:39:07 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-30 00:39:07 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-30 00:39:07 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.5',
  'to' => '3.12.0',
)]
2023-03-30 00:39:07 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.5',
  'to' => '3.12.0',
)]
2023-03-30 00:42:17 [info] elementor-pro::elementor_pro_updater Started 
2023-03-30 00:42:17 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-03-30 00:42:17 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-03-30 00:42:17 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.7',
  'to' => '3.12.0',
)]
2023-03-30 00:42:17 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.7',
  'to' => '3.12.0',
)]

PHP: showing 6 of 6PHP: 2023-01-21 23:20:14 [notice X 35][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined index: condition_type [array (
  'trace' => '
#0: /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-01-21 23:20:15 [notice X 2][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/loop-builder/module.php::84] Undefined index: elementor-preview [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-09 14:25:17 [notice X 9][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php::172] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-30 01:01:20 [error X 1][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on null in /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(13, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /Users/elewebdev/Local Sites/chat/app/public/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /Users/elewebdev/Local Sites/chat/app/public/wp [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-30 01:02:01 [notice X 6][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor/includes/managers/image.php::108] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-03-30 01:27:53 [error X 1][/Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::295] Uncaught Error: Call to a member function update_meta() on null in /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:295
Stack trace:
#0 /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(168): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(635, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /Users/elewebdev/Local Sites/chat/app/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /Users/elewebdev/Local Sites/chat/app/public/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /Users/elewebdev/Local Sites/chat/app/public/w [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2023-01-22 00:00:39 [error X 1][http://chat.local/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.10.1:3:1075047] Cannot read properties of null (reading 'getBoundingClientRect') 
JS: 2023-03-30 03:33:24 [error X 2][http://chat.local/wp-content/plugins/elementor/assets/js/df16c9d56464dd4f0800.bundle.min.js:2:7826] Cannot read properties of null (reading 'dataset') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by nicholaszein** [comment]
Hello @awakekat,

I can see you have activated the new Top Bar experiment. It's important to note that this experiment is an **Alpha** experiment and not all functionality is working at this point. It is there for testing purposes and should not be used on production websites.

If you deactivate this experiment, the Save as template function will work as expected.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor v3.13.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
